### PR TITLE
mp-b2c-carts-endpoints

### PR DESCRIPTION
**Pull request by shubham-khandelwal2-metacube**
## PR Description

https://spryker.atlassian.net/browse/CC-16138
mp-b2c-carts-endpoints-tests

### CODE REVIEW

File: resources/common/common_api.robot [hunk 1]
```diff
@@ -1245,6 +1245,27 @@ Find or create customer cart
         Run Keyword Unless    ${hasCart}    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}"}}}
         Run Keyword Unless    ${hasCart}    Save value to a variable:    [data][id]    cart_id
 
+
+Get ETag header value from cart
```
Comment (OKrivtsova) [inline resources/common/common_api.robot:1249]: please update description to specify that test scope variable is set (not just any variable)
also please re-generate the html documentation (see readme on how to)

File: tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    And Response should return error message:    Unauthorized request.
+
+Create_cart_with_invalid_access_token
+    When I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "${test_cart_name}"}}}
+    Then Response status code should be:    403
+    And Response reason should be:    Forbidden
+    And Response should return error message:    Missing access token.
+    And Response should return error code:    002
+
+Create_cart_when_cart_already_exist
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot:112]: FIXED

File: tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+Update_cart_without_cart_id
+      [Setup]    I set Headers:    Authorization=u2g3v4b6jk55b    If-Match=wrong_tag
+    When I send a PATCH request:    /carts/    {"data": {"type": "carts","attributes": {"priceMode": "${net_mode}","currency": "${currency_code_eur}","store": "${store_de}"}}}
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error message:    Resource id is not specified.
+
+
+Update_cart_with_invalid_header_tag
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot:158]: FIXED

File: tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    And Response body parameter should be:    [data][id]    ${cart_id}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    And Response body parameter should contain:    [data][attributes][totals]    expenseTotal
+    And Response body parameter should contain:    [data][attributes][totals]    discountTotal
+    And Response body parameter should contain:    [data][attributes][totals]    taxTotal
+    And Response body parameter should contain:    [data][attributes][totals]    subtotal
+    And Response body parameter should contain:    [data][attributes][totals]    grandTotal
+    And Response body parameter should contain:    [data][attributes][totals]    priceToPay
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot:29]: FIXED

File: tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    And Response body has correct self link internal
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot:211]: FIXED

File: tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+        ...  AND    I set Headers:    Authorization=${token}
+        ...  AND    Find or create customer cart
+        ...  AND    Save Header value to a variable:    ETag    header_tag
+        ...  AND    I set Headers:    Authorization=${token}    If-Match=${header_tag}
+    When I send a PATCH request:    /carts/${cart_id}    {"data": {"attributes": {"priceMode": "${net_mode}","currency": "${currency_code_eur}","store": "${store_de}"}}}
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error message:    Post data is invalid.
+
+Update_cart_with_invalid_priceMod_currency_store
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/negative.robot:201]: FIXED

File: tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+Test Setup        TestSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+    
+#GET requests
+Get_cart_by_cart_id
```
[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot:12]: DONE

File: resources/common/common_api.robot (hunk 1 above)
Comment (shubham-khandelwal2-metacube) [reply resources/common/common_api.robot:1249]: DONE

File: tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot (hunk 5 above)
[the remark this answers is not in the document]
Comment (OKrivtsova) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot:210]: not fully fixed.
the test needs to change mode no net mode - check that it was changed - and then return cart to the gross mode in teardown (like it is not in b2c tests)

[the remark this answers is not in the document]
Comment (shubham-khandelwal2-metacube) [reply tests/api/mp_b2c/glue/cart_endpoints/carts/positive.robot:210]: DONE

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**